modules/reporting/src/main/java/com/devonfw/module/reporting/common/api/Reporting.java

```diff
@@ -0,0 +1,74 @@
+package com.devonfw.module.reporting.common.api;
+
+import java.io.File;
+import java.io.OutputStream;
+import java.util.List;
+
+/**
+ * This is the interface for a simple facade to generate a report based on Jasper Reports library.
+ *
+ * @param <T> type of the data that is provided to be included in the report.
+ */
+public interface Reporting<T> {
+
+  /**
+   * Generates a report file with the given data, the template located in the templatePath and the type of file.
+   * specified.
+   *
+   * @param report the {@link Report} that encapsulates the data to be included in the report, the location of the
+   *        report template and the values of the parameters defined in the report template (in case template uses
+   *        parameters).
+   * @param file the file where the report must be created.
+   * @param format - report format
+   */
+  public void generateReport(Report report, File file, String format);
```

Comment:
* omit `public` keyword inside interface
* use default method delegating to generateReport with OutputStream?
* is `File` still the state of the art type of choice or should we support `java.nio.Path`?